src/posix/platform/netif.cpp

```diff
@@ -98,12 +98,24 @@ static void UpdateUnicast(otInstance *aInstance, const otIp6Address &aAddress, u
     ifr6.ifr6_ifindex   = static_cast<int>(sTunIndex);
     ifr6.ifr6_prefixlen = aPrefixLength;
 
-    VerifyOrExit(ioctl(sIpFd, (aIsAdded ? SIOCSIFADDR : SIOCDIFADDR), &ifr6) == 0, perror("ioctl");
-                 error = OT_ERROR_FAILED);
+    if (aIsAdded)
+    {
+        VerifyOrDie(ioctl(sIpFd, SIOCSIFADDR, &ifr6) == 0, OT_EXIT_ERROR_ERRNO);
+    }
+    else
+    {
+        VerifyOrExit(ioctl(sIpFd, SIOCDIFADDR, &ifr6) == 0, perror("ioctl"); error = OT_ERROR_FAILED);
+    }
 
 exit:
-    SuccessOrDie(error);
-    otLogInfoPlat("%s: %s", __func__, otThreadErrorToString(error));
+    if (error != OT_ERROR_NONE)
+    {
+        otLogInfoPlat("%s: %s", __func__, otThreadErrorToString(error));
```

Comment:
It seems INFO is used for the error case and WARN is used for success case. Is that intentional?